### PR TITLE
Add automatic configuration of LaTeXML server locally

### PR DESCRIPTION
**Pull request by plusparth**

### CODE REVIEW

File: manifests/latexml.pp [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    directory => '/vagrant/srv/latexml/ltxpsgi',
+  }
+  git::clone { 'search':
+    remote    => 'https://github.com/KWARC/LaTeXML-Plugin-MathWebSearch.git',
+    directory => '/vagrant/srv/latexml/search',
+  }
+ 
+
+  exec { 'latexml prerequisites':
+    command => '/usr/bin/apt-get install libarchive-zip-perl libfile-which-perl libimage-size-perl libio-string-perl libjson-xs-perl libwww-perl libparse-recdescent-perl liburi-perl libxml2 libxml-libxml-perl libxslt1.1 libxml-libxslt-perl texlive imagemagick perlmagick make'
```
Comment (physikerwelt) [inline manifests/latexml.pp]: put those packages to the package list line (l9)

File: manifests/latexml.pp [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  }
+
+
+  exec { 'install latexml':
+    command => 'perl Makefile.PL && make && make install',
```
Comment (physikerwelt) [inline manifests/latexml.pp:41]: What prevents this exec to run on every vagrant provision command? Normally you would have an creates=>

File: manifests/latexml.pp [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+  file { "/etc/apache2/sites-available/latexml.conf":
+    notify  => Service['apache2'],
+    ensure  => present,
+    content => template('drmf/latexml.conf.erb'),
+    require => Exec['install search'],
+  }
+
+
+  exec { 'deploy latexml':
```
Comment (physikerwelt) [inline manifests/latexml.pp:98]: why do you need that? I think there should be a service module, where you can require a service to be running.

File: manifests/latexml.pp [hunk 4]
```diff
@@ -0,0 +1,88 @@
+# == Class: drmf::latexml==
+class drmf::latexml(
+#   $M2_HOME = '/usr/share/maven',
```
Comment (physikerwelt) [inline manifests/latexml.pp]: remove the comments.

**Comment by physikerwelt** [comment]
nice job did you test it with a fresh vm.
To get a fresh vm
1. create dumps of all databases (i.e., wiki, enwiki, and wikidatawiki) via mysldump
2. run vagrant destroy
3. run vagrant up again

File: manifests/latexml.pp (hunk 3 above)
Comment (plusparth) [inline manifests/latexml.pp:98]: You can require the service to be running, but afaik you can't enable the site and then force service restart.

Comment (physikerwelt) [inline manifests/latexml.pp:98]: Did you read this http://www.puppetcookbook.com/posts/restart-a-service-when-a-file-changes.html?

**Comment by plusparth** [comment]
Is there some place the dumps should be stored so that it will automatically restore? Also, what type of file should it output to?

File: manifests/latexml.pp (hunk 3 above)
Comment (plusparth) [inline manifests/latexml.pp:98]: I did, but I didn't think of using that on this command - fixed now

**Comment by physikerwelt** [comment]
Did the clean run finally succeed?

**Comment by plusparth** [comment]
Now it sometimes fails to deploy mathosphere - it just says build failed.

**Comment by physikerwelt** [comment]
I'd suggest to split this into two pull request. One for latexml and one for mathosphere.

**Comment by plusparth** [comment]
@physikerwelt This is ready to be merged (it's been done for a while)

**Pull request merged**